test-nokogumbo.rb

```diff
@@ -137,6 +138,22 @@ def test_parse_fragment_errors
     refute_empty doc.errors
   end
 
+  def test_document_encoding
+    html = <<-TEXT
+      <html>
+        <head>
+          <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+        </head>
+        <body>
+          Кирилические символы
+        </body>
+      </html>
+    TEXT
+    doc = Nokogiri::HTML5.parse(html)
+    assert_equal "UTF-8", doc.encoding
+    assert_equal "Кирилические символы", doc.at('body').text.gsub(/\n\s+/,'')
+  end
```

Comment:
The second assertion isn't related to the encoding. I think the test should be something like

``` ruby
def test_document_encoding
  doc = Nokogiri::HTML5.parse(buffer)
  assert_equal 'UTF-8', doc.encoding
```